### PR TITLE
Closes #432 Update Elide to version 5

### LINKED ISSUE #432: Update Elide to version 5
https://elide.io/pages/guide/v5/17-migration.html

### PR DESCRIPTION
**Pull request by bukajsytlos**
Closes #432

### CODE REVIEW

File: src/main/java/com/faforever/api/config/elide/SpringHibernateDataStore.java [hunk 1]
```diff
@@ -118,10 +118,6 @@ public void populateEntityDictionary(EntityDictionary dictionary) {
           dictionary.lookupEntityClass(mappedClass);
           // Bind if successful
           dictionary.bindEntity(mappedClass);
-          if (isSpringDependencyInjection) {
-            // Bind Spring Dependency Injection
-            dictionary.bindInitializer(beanFactory::autowireBean, mappedClass);
-          }
```
Comment (Brutus5000) [inline src/main/java/com/faforever/api/config/elide/SpringHibernateDataStore.java:124]: From my understanding the beanFactory is no longer used in this component and should be removed

File: src/main/java/com/faforever/api/config/elide/ElideConfig.java [hunk 2]
```diff
@@ -102,7 +103,7 @@ private void registerAdditionalConverters() {
   }
 
   @Bean
-  public EntityDictionary entityDictionary() {
+  public EntityDictionary entityDictionary(AutowireCapableBeanFactory beanFactory) {
     ConcurrentHashMap<String, Class<? extends Check>> checks = new ConcurrentHashMap<>();
```
Comment (Brutus5000) [inline src/main/java/com/faforever/api/config/elide/ElideConfig.java]: I read something about checks can be found via annotation now. maybe that would be interesting for us?

File: src/main/java/com/faforever/api/data/DataController.java [hunk 3]
```diff
@@ -74,7 +79,9 @@ private static Object getPrincipal(final Authentication authentication) {
       getJsonApiPath(request),
       body,
       new MultivaluedHashMap<>(allRequestParams),
-      getPrincipal(authentication)
+      getPrincipal(authentication),
+      API_VERSION,
+      UUID.randomUUID()
```
Comment (Brutus5000) [inline src/main/java/com/faforever/api/data/DataController.java:84]: What is this random uuid doing? Does it make sense to put something else in there, e.g. the inbound request id?

File: src/main/java/com/faforever/api/data/checks/IsEntityOwner.java [hunk 4]
```diff
@@ -1,30 +1,25 @@
...
-        && entity.getEntityOwner() != null
-        && entity.getEntityOwner().getId() == ((FafUserDetails) opaqueUser).getId();
-    }
-
-    @Override
-    public boolean ok(com.yahoo.elide.security.User user) {
-      return false;
+      final ElideUser caller = (ElideUser) requestScope.getUser();
+      return entity.getEntityOwner() != null
+        && entity.getEntityOwner().getId().equals(caller.getFafUserDetails().map(FafUserDetails::getId).orElse(null));
```
Comment (Brutus5000) [inline src/main/java/com/faforever/api/data/checks/IsEntityOwner.java:24]: Maybe Object.equals instead? More readable, slightly more null safe.

**Review by Brutus5000 — COMMENTED** [review]
I'm so excited about the final release

File: src/main/java/com/faforever/api/data/checks/IsEntityOwner.java (hunk 4 above)
Comment (bukajsytlos) [reply src/main/java/com/faforever/api/data/checks/IsEntityOwner.java:24]: I don't want to null == null be true

File: src/main/java/com/faforever/api/data/DataController.java (hunk 3 above)
Comment (bukajsytlos) [reply src/main/java/com/faforever/api/data/DataController.java:84]: yes I guess so, do we have request id in some header?

Comment (bukajsytlos) [reply src/main/java/com/faforever/api/data/DataController.java:84]: it looks like to be used to bound transaction to this id and then I guess it's used in async calls.
nothing to do with logging/audit

File: src/main/java/com/faforever/api/config/elide/SpringHibernateDataStore.java (hunk 1 above)
Comment (bukajsytlos) [reply src/main/java/com/faforever/api/config/elide/SpringHibernateDataStore.java:124]: you are right

File: src/main/java/com/faforever/api/config/elide/ElideConfig.java (hunk 2 above)
Comment (bukajsytlos) [reply src/main/java/com/faforever/api/config/elide/ElideConfig.java]: why not :)

**Comment by Brutus5000** [comment]
Screw it, we're gonna ship it with RC4! :smile:

**Pull request merged**